app/Models/UserGroupEvent.php

```diff
@@ -102,18 +116,42 @@ private static function log(?User $actor, string $type, ?User $user, Group $grou
         ));
     }
 
-    public function actor()
+    public function actor(): BelongsTo
     {
         return $this->belongsTo(User::class, 'actor_id');
     }
 
-    public function group()
+    public function group(): BelongsTo
     {
         return $this->belongsTo(Group::class, 'group_id');
     }
 
-    public function user()
+    public function user(): BelongsTo
     {
         return $this->belongsTo(User::class, 'user_id');
     }
+
+    public function getAttribute($key)
+    {
+        return match ($key) {
+            'actor_id',
+            'group_id',
+            'id',
+            'type',
+            'user_id' => $this->getRawAttribute($key),
+
+            'details' => json_decode($this->getRawAttribute($key), true),
```

Comment:
the column is nullable and json_decode will throw error here because it requires string